core/src/main/java/de/danoeh/antennapod/core/cast/RemoteMedia.java

```diff
@@ -0,0 +1,358 @@
+        }
+        if (!TextUtils.isEmpty(itemIdentifier)) {
+            metadata.putString(CastUtils.KEY_EPISODE_IDENTIFIER, itemIdentifier);
+        } else {
+            metadata.putString(CastUtils.KEY_EPISODE_IDENTIFIER, downloadUrl);
+        }
+        if (!TextUtils.isEmpty(episodeLink)) {
+            metadata.putString(CastUtils.KEY_EPISODE_LINK, episodeLink);
+        }
+        String notes = this.notes;
+        if (notes != null) {
+            if (notes.length() > CastUtils.EPISODE_NOTES_MAX_LENGTH) {
+                notes = notes.substring(0, CastUtils.EPISODE_NOTES_MAX_LENGTH);
+            }
+            metadata.putString(CastUtils.KEY_EPISODE_NOTES, notes);
+        }
+        // Default id value
+        metadata.putInt(CastUtils.KEY_MEDIA_ID, 0);
+        metadata.putInt(CastUtils.KEY_FORMAT_VERSION, CastUtils.FORMAT_VERSION_VALUE);
+
+        MediaInfo.Builder builder = new MediaInfo.Builder(downloadUrl)
+                .setContentType(mime_type)
+                .setStreamType(MediaInfo.STREAM_TYPE_BUFFERED)
+                .setMetadata(metadata);
+        if (duration > 0) {
+            builder.setStreamDuration(duration);
+        }
+        return builder.build();
+    }
+
+    public String getEpisodeIdentifier() {
+        return itemIdentifier;
+    }
+
+    public String getFeedUrl() {
+        return feedUrl;
+    }
+
+    public FeedMedia lookForFeedMedia() {
+        FeedItem feedItem = DBReader.getFeedItem(feedUrl, itemIdentifier);
+        if (feedItem == null) {
+            return null;
+        }
+        return feedItem.getMedia();
+    }
+
+    @Override
+    public void writeToPreferences(SharedPreferences.Editor prefEditor) {
+        //it seems pointless to do it, since the session should be kept by the remote device.
+    }
+
+    @Override
+    public void loadMetadata() throws PlayableException {
+        //Already loaded
+    }
+
+    @Override
+    public void loadChapterMarks() {
+        ChapterUtils.loadChaptersFromStreamUrl(this);
+    }
+
+    @Override
+    public String getEpisodeTitle() {
+        return episodeTitle;
+    }
+
+    @Override
+    public List<Chapter> getChapters() {
+        return chapters;
+    }
+
+    @Override
+    public String getWebsiteLink() {
+        if (episodeLink != null) {
+            return episodeLink;
+        } else {
+            return feedUrl;
+        }
+    }
+
+    @Override
+    public String getPaymentLink() {
+        return null;
+    }
+
+    @Override
+    public String getFeedTitle() {
+        return feedTitle;
+    }
+
+    @Override
+    public Object getIdentifier() {
+        return itemIdentifier + "@" + feedUrl;
+    }
+
+    @Override
+    public int getDuration() {
+        return duration;
+    }
+
+    @Override
+    public int getPosition() {
+        return position;
+    }
+
+    @Override
+    public long getLastPlayedTime() {
+        return lastPlayedTime;
+    }
+
+    @Override
+    public MediaType getMediaType() {
```

Comment:
I only suggest this since this conversion is now done in a few different places. If we ever add some other type we'd have to change it everywhere or else have strange errors.